src/main/java/software/amazon/awssdk/crt/mqtt5/Mqtt5ListenerOptions.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.mqtt5;
+
+import software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions.LifecycleEvents;
+
+/**
+ * Configuration for the creation of Mqtt5Listener
+ *
+ * MQTT5 support is currently in <b>developer preview</b>.  We encourage feedback at all times, but feedback during the
+ * preview window is especially valuable in shaping the final product.  During the preview period we may make
+ * backwards-incompatible changes to the public API, but in general, this is something we will try our best to avoid.
+ */
+public class Mqtt5ListenerOptions {
+
+    private LifecycleEvents lifecycleEvents;
+    private ListenerPublishEvents listenerPublishEvents;
+
+    /**
+     * Returns the LifecycleEvents interface that will be called when the client gets a LifecycleEvent.
+     *
+     * @return The LifecycleEvents interface that will be called when the client gets a LifecycleEvent
+     */
+    public LifecycleEvents getLifecycleEvents() {
+        return this.lifecycleEvents;
+    }
+
+    /**
+     * Returns the PublishEvents interface that will be called when the client gets a message.
+     *
+     * @return PublishEvents interface that will be called when the client gets a message.
+     */
+    public ListenerPublishEvents getListenerPublishEvents() {
+        return this.listenerPublishEvents;
+    }
+
+    /*******************************************************************************
+     * callback methods
+     ******************************************************************************/
+
+    /**
+     * An interface that defines all of the publish functions the Mqtt5Client will call when it receives a publish packet.
+     */
+    public interface ListenerPublishEvents {
```

Comment:
I kept both interfaces because the two callbacks have different behaviors. ListenerPublishEvents is only for the listener. If the `ListenerPublishEvents` returned true, which means the `publishReceivedEvent` get handled, then it would skip the Client's `PublishEvents` callback. While the Client's original `PublishEvents` interface would never do that. That's why the original `PublishEvents` returns `void` instead of `bool`. 
I agreed that the two interfaces with similar names are confusing. Let me think if there is a better way to handle the interface.

Reply:
Ah I see, so the `ListenerPublishEvents` is just for the listener(s) while `PublishEvents` would be more a MQTT5 client-wide thing that only gets invoked if no listeners have not snatched the publish already.

I think I was just misunderstanding on my part. I thought they were essentially the same interface and functionality, just the listener needed a boolean for additional processing but in theory the `PublishEvents` could return a boolean too for the some functionality should it be exposed. Since they are completely separate and the `PublishEvents` is essentially the final stop if no listeners have processed it, then I think it's probably fine. Apologizes for the misunderstanding on my part 😅 

The naming is probably okay then, I would just document on both `ListenerPublishEvents` and `PublishEvents` how `ListenerPublishEvents` will get the message/publish first, and if it marks the message as processed then the `PublishEvents` associated with the MQTT5 client will not get the message. Something like that I think would clear it up any confusion should anyone be confused and looking at the interfaces.